### ISSUE TITLE
[@types/hapi__hapi] Needs to be updated bc of Podium breaking changes

### ISSUE THREAD

**Issue by nicoleczhu** [DefinitelyTyped/DefinitelyTyped#52135]
Is something wrong with CI or publishing infrastructure?

- Check the [Infrastructure status updates](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44317) thread for known issues.
- Leave a message on the [DefinitelyTyped Gitter](http://gitter.im/DefinitelyTyped/DefinitelyTyped).

If you know how to fix the issue, make a pull request instead.

- [ x ] I tried using the `@types/xxxx` package and had problems.
- [ x ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ x ] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [ x ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ x ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @rafaelsouzaf , @jhsimms , @SimonSchick , @saboya , @lenovouser

If you do not mention the authors the issue will be ignored.

### The problem
Podium release [4.1.2](https://www.npmjs.com/package/@hapi/podium) breaks this module:

```sh

node_modules/@types/hapi__hapi/index.d.ts:260:18 - error TS2430: Interface 'RequestEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish" | "disconnect", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

260 export interface RequestEvents extends Podium {
                     ~~~~~~~~~~~~~

node_modules/@types/hapi__hapi/index.d.ts:633:18 - error TS2430: Interface 'ResponseEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

633 export interface ResponseEvents extends Podium {
                     ~~~~~~~~~~~~~~

node_modules/@types/hapi__hapi/index.d.ts:2363:43 - error TS2314: Generic type 'Listener' requires 1 type argument(s).

2363     removeListener(criteria: K, listener: Podium.Listener): this;
                                               ~~~~~~~~~~~~~~~

node_modules/@types/hapi__hapi/index.d.ts:2381:18 - error TS2430: Interface 'ServerEvents' incorrectly extends interface 'Podium'.
  The types returned by 'on(...)' are incompatible between these types.
    Type 'void' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'void'.

2381 export interface ServerEvents extends Podium {
                      ~~~~~~~~~~~~

node_modules/@types/hapi__hapi/index.d.ts:2447:44 - error TS2314: Generic type 'Listener' requires 1 type argument(s).

2447     removeListener(name: string, listener: Podium.Listener): Podium;
           
```

**Comment by saboya** [comment]
After going through the new typings a bit, I believe they are just wrong.

If you check Podium´s source code:

https://github.com/hapijs/podium/blob/5b81f6f4cd98d7271f4c2ae655b7776d87c24a08/lib/index.js#L211-L230

You can see that, if the `spread` flag is passed when creating the event, Podium will spread the event arguments over the callback function when invoking it. However, the new typings lister doesn't account for that:

https://github.com/hapijs/podium/blob/5b81f6f4cd98d7271f4c2ae655b7776d87c24a08/lib/index.d.ts#L197-L198

Considering how the current hapi typings is written, I can´t see why any changes would be necessary, the new typings are just wrong, unless I´m missing something.

**Comment by saboya** [comment]
This is currently being worked on: https://github.com/hapijs/podium/pull/70

As soon as the fix lands, we can do the necessary changes to work with the new typings.

**Comment by ceisele-r** [comment]
@saboya it looks like https://github.com/hapijs/podium/pull/70 was released in @hapi/podium `v4.1.3`
